src/vcpkg/commands.build.cpp

```diff
@@ -666,14 +783,60 @@ namespace vcpkg
                                            const PreBuildInfo& pre_build_info,
                                            const Toolset& toolset)
     {
+        static constexpr auto vcpkg_json = R"--(
```

Comment:
Now that we have https://github.com/microsoft/vcpkg/pull/36398 do you mind if I replace this attempt to move detect_compiler out with just using what @data-queue added?

Reply:
You can do it :)